js/gdm/loginDialog.js

```diff
@@ -37,9 +37,9 @@ const Batch = imports.gdm.batch;
 const BoxPointer = imports.ui.boxpointer;
 const CtrlAltTab = imports.ui.ctrlAltTab;
 const GdmUtil = imports.gdm.util;
-const Layout = imports.ui.layout;
 const LoginManager = imports.misc.loginManager;
 const Main = imports.ui.main;
+const Monitor = imports.ui.monitor;
```

Comment:
We need this ourselves or the shell will break because forceAppExitDialog.js, which imports modalDialog.js, gets imported from windowManager.js, which in turn gets (recursively) imported from modalDialog.js too... causing the issue.

So, unless I can make this forceAppExitDialog work by importing it from somewhere else, I'm afraid we'd need this change downstream. I've tried to convince Florian upstream to get it included, but he has a good point on that, without anything upstream actually needing it, it seems a random change to make that, on top of it, would break extensions unless we added a small compatibility hack.